File: generator/src/test/resources/swift/expected/RequestQueryParamsTest/test-optional-query-parameter-generation.output.swift
import Sunday

public class API {

  public let requestFactory: RequestFactory
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    requestFactory: RequestFactory,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json]
  ) {
    self.requestFactory = requestFactory
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
  }

  public func fetchTest(
    obj: Test? = nil,
    str: String? = nil,
    int: Int? = nil,
    def1: String? = "test",
    def2: Int? = 10
  ) async throws -> Test {
    return try await self.requestFactory.result(
      method: .get,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: [
        "obj": obj as Any?,
        "str": str as Any?,
        "int": int as Any?,
        "def1": def1 as Any?,
        "def2": def2 as Any?
      ].filter { $0.value != nil },
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

}

File: generator/src/test/resources/kotlin/expected/sunday/RequestQueryParamsTest/test-optional-query-parameter-generation.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.Operation
import io.outfoxx.sunday.OperationSpec
import io.outfoxx.sunday.Transport
import io.outfoxx.sunday.http.Method
import io.outfoxx.sunday.http.Request
import io.outfoxx.sunday.operation
import io.test.Test
import kotlin.Int
import kotlin.String
import kotlin.Unit
import kotlin.collections.List

public class API<Req : Request>(
  public val transport: Transport<Req>,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public fun fetchTest(
    obj: Test? = null,
    str: String? = null,
    int: Int? = null,
    def1: String? = "test",
    def2: Int? = 10,
  ): Operation<Unit, Test, Req> = this.transport.operation<Unit, Test, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/tests",
      queryParameters = mapOf(
        "obj" to obj,
        "str" to str,
        "int" to int,
        "def1" to def1,
        "def2" to def2
      ).filterValues { it != null },
      acceptTypes = this.defaultAcceptTypes
    )
  )
}
